compiler/proto.h

```diff
@@ -359,10 +359,13 @@ extern void size_type P_ ((TYP *));
 #undef malloc
 #undef realloc
 
-#define	free(s)			fmemfree(s)
-#define	malloc(s)		fmemalloc(s)
-#define realloc(p,s)    fmemrealloc((p),(s))
+#define	free(s)			memfree(s)
+#define	malloc(s)		memalloc(s)
+#define realloc(p,s)    memrealloc((p),(s))
 
+void __far *memalloc(unsigned long size);
+void memfree(void __far *ptr);
+void __far *memrealloc(void __far *ptr, unsigned long size);
 #endif
```

Comment:
This looks great, but perhaps consider putting the undef's, define's and C declarations all in mem.h and including mem.h into proto.h. That separates out the external API and then only a single file need be changed when the functions are moved into the C library: mem.h and mem.c (both would be deleted in the applications, as they would be replaced by C library code or standard headers).